### PR TITLE
Modify lib/mm_network.pm and use for tests/network/firewalld_policy_objects.pm use

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#105295](https://progress.opensuse.org/issues/105295)
- Failure being fixed: [firewalld-policy-firewall](https://openqa.opensuse.org/tests/2303282)
- Previous pull request: #14479
- Verification run: See [wicked@Tumbleweed](http://pdostal-server.suse.cz/tests/14636), [Wicked@SLE15-SP3](http://pdostal-server.suse.cz/tests/14638), [wicked@SLE12-SP4](http://pdostal-server.suse.cz/tests/14641), [wireguard@Tumbleweed](http://pdostal-server.suse.cz/tests/14630), [firewalld@Tumbleweed](http://pdostal-server.suse.cz/tests/14633)

Signed-off-by: Pavel Dostál <pdostal@suse.cz>

### CODE REVIEW

File: lib/mm_network.pm [hunk 1]
```diff
@@ -103,7 +108,21 @@ sub configure_dhcp {
...
+    if ($is_nm) {
+        my $nm_id;
+        my $nm_list = script_output("nmcli -t -f DEVICE,NAME c | grep '$device' | head -n1");
+        ($device, $nm_id) = split(':', $nm_list);
+
+        assert_script_run "nmcli connection modify '$nm_id' ipv4.gateway 10.0.2.2";
+        assert_script_run "nmcli connection down '$nm_id'";
+        assert_script_run "nmcli connection up '$nm_id'";
+    } else {
+        enter_cmd("echo 'default 10.0.2.2 - -' > /etc/sysconfig/network/routes");
```
Comment (cfconrad) [inline lib/mm_network.pm:123]: This doesn't change the running network configuration within wicked.
A `wicked ifup all` should do it.
```suggestion
        enter_cmd("echo 'default 10.0.2.2 - -' > /etc/sysconfig/network/routes");
        enter_cmd('wicked ifup all');
```

File: lib/mm_network.pm [hunk 2]
```diff
@@ -65,24 +71,23 @@ sub configure_static_ip {
...
 
         # Restart the networking
         assert_script_run "rcnetwork restart";
     }
 
     save_screenshot;
-    assert_script_run "ip addr";
+    assert_script_run "ip addres show";
+    assert_script_run "ip route show";
     save_screenshot;
```
Comment (cfconrad) [inline lib/mm_network.pm:91]: record_info('network cfg', script_output('ip a s; ip r s'));

File: lib/mm_network.pm [hunk 3]
```diff
@@ -65,24 +71,23 @@ sub configure_static_ip {
...
 
         # check for duplicate IP
         my ($ip_no_mask, $mask) = split('/', $ip);
-        script_run "arping -w 1 -I \$NIC $ip_no_mask";
+        script_run "arping -w 1 -I $device $ip_no_mask";
 
         # Configure the static networking
-        assert_script_run "echo -e \"STARTMODE='auto'\\nBOOTPROTO='static'\\nIPADDR='$ip'\\nMTU='$mtu'\" > /etc/sysconfig/network/ifcfg-\$NIC";
-
-        configure_default_gateway();
```
[the remark this answers is not in the document]
Comment (cfconrad) [reply lib/mm_network.pm:78]: Hm now I see that it was moved around here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14479/files#diff-3abd5bd42de4f5a93e446d50190f950ef242424baa868c8c17241ba6e7a32902L180 not sure now what is the right solution. Maybe adding it in back in `setup_static_mm_network()` should solve it.

File: lib/mm_network.pm [hunk 4]
```diff
@@ -216,7 +218,23 @@ sub setup_static_mm_network {
...
+    my $is_nm = $args{is_nm} // is_networkmanager();
+
+    if ($is_nm) {
+        assert_script_run 'nmcli networking off';
+        assert_script_run 'nmcli networking on';
+    } else {
+        assert_script_run 'wicked ifup all';
+    }
+
+    record_info('network cfg', script_output('ip addres show; echo; ip route show; cat /etc/resolv.conf'));
```
Comment (Vogtinator) [inline lib/mm_network.pm:237]: IMO it should either be `ip a s` or the full form `ip address show`

```suggestion
    record_info('network cfg', script_output('ip address show; echo; ip route show; cat /etc/resolv.conf'));
```

File: lib/mm_network.pm (hunk 3 above)
[the remark this answers is not in the document]
Comment (pdostal) [reply lib/mm_network.pm:78]: I manually checked that everywhere in `lib/ tests/ products/*/main.pm` where `configure_static_ip` is mentioned also `configure_default_gateway` is.

File: lib/mm_network.pm [hunk 5]
```diff
@@ -216,7 +218,23 @@ sub setup_static_mm_network {
...
+
+sub restart_networking {
+    my (%args) = @_;
+    my $is_nm = $args{is_nm} // is_networkmanager();
+
+    if ($is_nm) {
+        assert_script_run 'nmcli networking off';
+        assert_script_run 'nmcli networking on';
+    } else {
+        assert_script_run 'wicked ifup all';
```
Comment (cfconrad) [inline lib/mm_network.pm:234]:
```suggestion
        assert_script_run 'wicked ifdown all';
        assert_script_run 'wicked ifup all';
```
I think this fit's more.

**Review by cfconrad — APPROVED** [review]
LGTM, only small hint regarding wicked, which I think make it more reliable.

**Review by jlausuch — APPROVED** [review]

File: lib/mm_network.pm [hunk 6]
```diff
@@ -216,7 +218,24 @@ sub setup_static_mm_network {
     my $ip = shift;
     my $is_nm = is_networkmanager();
     configure_static_ip(ip => $ip, is_nm => $is_nm);
+    configure_default_gateway(is_nm => $is_nm);
```
[the remark this answers is not in the document]
Comment (pdostal) [reply lib/mm_network.pm:221]: It's found automatically. NetworkManager needs connection name, not device. When it's not specified, the first is used. That's also how it was before. Please see the subroutine.

[the remark this answers is not in the document]
Comment (Vogtinator) [reply lib/mm_network.pm:221]: I did, it does:

```
    my $device = $args{device};
    if ($is_nm) {
        my $nm_id;
        my $nm_list = script_output("nmcli -t -f DEVICE,NAME c | grep '$device' | head -n1");
```

i.e. it unconditionally uses `$device` and it's not set here.

[the remark this answers is not in the document]
Comment (pdostal) [reply lib/mm_network.pm:221]: It's also not specified as by default SUT has only one interface so one connection. The loopback is not being controlled by NetworkManager.

[the remark this answers is not in the document]
Comment (pdostal) [reply lib/mm_network.pm:221]: Yes, when `$device` is empty then `grep` basically does nothing. Should I put a comment over there about it?

[the remark this answers is not in the document]
Comment (Vogtinator) [reply lib/mm_network.pm:221]: Absolutely, currently it looks like a bug.

[the remark this answers is not in the document]
Comment (pdostal) [reply lib/mm_network.pm:222]: Done. Is this resolved?

[the remark this answers is not in the document]
Comment (Vogtinator) [reply lib/mm_network.pm:222]: Yep

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/2325801#step/setup/45 seems to be a fallout from here?

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/2326410#step/xrdp_client/15 this somewhat also seems related to this change

**Comment by pdostal** [comment]
> https://openqa.opensuse.org/tests/2326410#step/xrdp_client/15 this somewhat also seems related to this change

Please give me some time (one day?) and I will submit another small fix for this.

**Pull request merged**